### DISCUSSION TITLE
appendix_problem

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/labview_book#55] [Announcements]
# appendix_problem

这一节，记录笔者在使用 LabVIEW 过程中遇到的一些问题，以及 LabVIEW 一些不太完善的方面。

https://lv.qizhen.xyz/appendix_problem

**Comment by lhb5883** [comment]
现在Windows10和Windows11为了改善和WLS的互操作性，在控制面板的区域语言选项已经有了测试版的UTF-8支持，在Control Panel\Clock and Region\Region\Administrative\Language for non-Unicode programs\Change System Locale\Beta:Use Unicode UTF-8 for worldwide language support.可以尝试看看能不能生效，缺点是这个副作用是全局的，会影响所有非Unicode程序。如果生效，可以考虑用NTLEA或者其他第三方工具单独设置LabVIEW。

**Reply by lhb5883** [reply]
测试了一下，启用了Beta:Use Unicode UTF-8 for worldwide language support之后，LabVIEW打开Linux下面编写的VI可以正常显示中文，但是副作用是LabVIEW里面字体列表中文的字体名字编程乱码了，另外Windows上没有文泉驿，fallback的字体显示偏小。

**Reply by lhb5883** [reply]
目前测试了一下，可以利用微软官方的Windows SDK给LabVIEW签名，测试是可以单独给LabVIEW启用UTF-8支持的具体方法参考
https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page
用的是LabVIEW 2023Q3 64bit。

**Reply by samthephantom** [reply]
您好，我们最近有一个以前写的labview程序要迁移到ubuntu里面编译和运行，请问这个方法会给labview单独启用UTF-8支持后出现中文乱码吗

**Reply by ruanqizhen** [reply]
我没做过实验，但如果是把以前写的 LabVIEW 程序改换文字编码，我觉得是会出现乱码的。

**Comment by swordscan** [comment]
本来还想说既然labview用操作系统默认编码，那么我就直接把win的默认编码改为utf-8，这样win和linux就能互通了。然后才看到楼上兄弟的实践，看来是可行啊。

**Reply by ruanqizhen** [reply]
可行，但是有代价：主要是兼容性的问题，如果需要使用以前编写好的程序，或者使用别人写好的程序，就要考虑那些程序是在什么系统下写的。或者你写的程序，别人还能用起来不。

**Comment by zhangcheng02** [comment]
请问能出一期labview使用NI采集卡的教程吗？

**Reply by ruanqizhen** [reply]
这一点我恐怕无能为力了。我目前的工作与 LabVIEW 和测试采集等都不相关。有时候利用免费版的 LabVIEW 学习一下，但采集卡根本接触不到了。

**Comment by zhangcheng02** [comment]
好的，非常感谢，您的教程很有意义！
        
            
        
    
         
        
            
                        ***@***.***
            
        
     
        
  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        12/13/2023 21:32
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [ruanqizhen/labview_book] appendix_problem (Discussion #55)
        
      
    
  

这一点我恐怕无能为力了。我目前的工作与 LabVIEW 和测试采集等都不相关。有时候利用免费版的 LabVIEW 学习一下，但采集卡根本接触不到了。
